### PR TITLE
Webpack v5

### LINKED ISSUE #10107: Webpack v5
Let's upgrade webpack to v5.

For QA: this affects the entire site but hopefully it shouldn't have an impact. Style and images should load fine and the site should work as before, there shouldn't be any noticeable changes.

---

### `yarn amo:dev`

- [x] investigate the following warning:

      replace autoprefixer browsers option to Browserslist config

- [x] investigate the following error:

      [DEP_WEBPACK_TEMPLATE_PATH_PLUGIN_REPLACE_PATH_VARIABLES_HASH]
      DeprecationWarning: [hash] is now [fullhash] (also consider using
      [chunkhash] or [contenthash], see documentation for details)
      [0] (Use `node --trace-deprecation ...` to show where the warning
      was created)

- [x] investigate the following error:

      [webpack-isomorphic-tools/plugin] [error] Module
      "./src/amo/img/icons/default-64.png" has no source. Maybe Webpack
      compilation of this module failed. Skipping this asset.

### PR DESCRIPTION
**Pull request by willdurand**
Fixes #10107

---

This has lots of changes but everything seems to work fine.

### CODE REVIEW

File: webpack.prod.config.babel.js [hunk 1]
```diff
@@ -58,34 +58,17 @@ export default {
...
-      new OptimizeCssAssetsPlugin({
-        cssProcessorPluginOptions: {
-          preset: [
-            'default',
-            {
-              svgo: {
-                plugins: [
-                  {
-                    // There is a bug in this optimization.
-                    // See https://github.com/mozilla/addons-frontend/issues/7191
```
Comment (willdurand) [inline webpack.prod.config.babel.js:70]: looks fixed to me

**Comment by willdurand** [comment]
I filed https://github.com/mozilla/addons-frontend/issues/10152 from some more warnings/errors in the _prod_ build output. It was there before, I'll investigate later as it doesn't seem to cause any issue.

**Review by bobsilverberg — APPROVED** [review]
Looks good, and not as many changes as I'd thought. Thanks for digging into this @willdurand!

I tried it locally and I don't see any problems. I saw those warnings you mentioned in slack about size limits, and I also see some error output from `webpack-isomorphic-tools/plugin`, but it doesn't seem to affect the build. Example error:

```
[webpack-isomorphic-tools/plugin] [error] Module "css ./node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[1].use[1]!./node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[1].use[2]!./node_modules/sass-loader/dist/cjs.js!./src/amo/fonts/fira.scss" has no source. Maybe Webpack compilation of this module failed. Skipping this asset.
```

I added one comment about pinning eslint.

r+wc

File: tests/unit/test_packageJson.js [hunk 2]
```diff
@@ -20,7 +20,17 @@ const packageJson = JSON.parse(
...
 // https://github.com/eslint/eslint/issues/12117
-const skipDevDeps = ['prettier', 'pretty-quick', 'enzyme', 'eslint'];
+
+// html-webpack-plugin and webpack-subresource-integrity: we're using unstable
+// versions.
+const skipDevDeps = [
+  'prettier',
+  'pretty-quick',
+  'enzyme',
+  'eslint',
```
[the remark this answers is not in the document]
Comment (willdurand) [reply tests/unit/test_packageJson.js:30]: oh right, I'll update that.

**Pull request merged**